Add Symfony 6.x support

rvanlaak opened the pull request:
No BC breaks have been found on the related Symfony packages:

* http-kernel https://github.com/symfony/http-kernel/blob/6.2/CHANGELOG.md#60
* messenger https://github.com/symfony/messenger/blob/6.2/CHANGELOG.md#60

Commit:
Add Symfony 6.x support

rvanlaak commented:
@janvernieuwe @syrm @jvocampings could you help with running the workflows, reviewing the PR and merging & tagging a new release? 

Looking forward to taking this bundle into production that excels in its simplicity! Thanks!

janvernieuwe commented:
Looks good to me! will merge after my vacation. Feel free to ping me again in case I forget in like a week or so.

rvanlaak commented:
Could you maybe approve the CI workflow from your hopefully beautiful holiday spot?

janvernieuwe committed:
Update CI config

janvernieuwe committed:
Update CI config

janvernieuwe committed:
Up phpcsfixer dependency

janvernieuwe commented:
@rvanlaak wasn't watching my mails on holidays... It was a little more work then just pushing the button.
I updated the CI config so it runs. Tests needs some updates with final classes.

Thanks again for the PR!

janvernieuwe committed:
Remove lint check

rvanlaak commented on .php-cs-fixer.dist.php:
```diff
@@ -0,0 +1,84 @@
+    ->setRules([
+        '@PSR12' => true,
+        '@Symfony' => true,
+        'align_multiline_comment' => true,
```
Oh wow, huge but nice list of custom cs rules. What about choosing `@Symfony:risky` instead? That might replace most of these rules?

janvernieuwe replied:
I took the list from our internal projects, so that everything is the same formatting. You can use auto fix tho...

Commit:
Fix code style

Commit:
Fix tests

rvanlaak commented:
Fixed code style in [817993c](https://github.com/ArxusPublic/messenger-newrelic/pull/5/commits/817993c70809b889d472d3354d8a3c09040ee44e)
Fixed tests in [f3d05ab](https://github.com/ArxusPublic/messenger-newrelic/pull/5/commits/f3d05ab808436815fd0d7f579f0e21c540d86743)

Workflows are waiting approval, so back to you!

rvanlaak commented:
@janvernieuwe available for workflow approval / code review? Am eager to try using this bundle in production asap.

janvernieuwe commented:
> @janvernieuwe available for workflow approval / code review? Am eager to try using this bundle in production asap.

Sorry that it took a while I didn't see it... feel free to give me a mention next time to push the button! 
Can you take a look at the windows failure?

rvanlaak commented:
@janvernieuwe I have no idea how the failing tests can be related to adding the Symfony 6 support. 

> PHP Fatal error:  Uncaught TypeError: fclose(): supplied resource is not a valid stream resource

They likely are there because of your changes to the CI. Can you fix those?

janvernieuwe commented:
We did some tests on a local windows machine here and they were fine (the phpunit at least).
I'll merge it and we'll take another look at the windows CI another time.

Thanks for the PR!

rvanlaak commented:
Am running the bundle in production now, on Symfony 6.1 and so far looking good.

One weird side effect that seems to somehow be happening (I've also enabled Monolog's `NewRelicHandler`), is that for serving huge files (read: file size > php memory limit) using `readfile` we started seeing OOM errors again (although having [this ob flush fix in place](https://stackoverflow.com/questions/31277672/allowed-memory-exhausted-when-using-readfile)).
